### PR TITLE
Comparison fixes

### PR DESCRIPTION
**Pull request by mxmeinhold**
### What?
Adds `__hash__()`, `__eq__()`, and `__repr__()` to CSHMember objects.

### Why?
Without `__eq__()` things like [list or set differencing](https://github.com/ComputerScienceHouse/conditional/blob/e3dab17eb4495e2177852a4988e20405ee04220c/conditional/util/member.py#L36) don't work, and `__hash__()` goes along with `__eq__()`.

Also adds `__repr__()` because I find dn to be more helpful than the memory address of a CSHMember instance, along the lines of #22.

### Caveats
I'm running python 3.8
* Python 3.7 and 3.8 added to travis.yml
* Pylint bumped to 2.5.2
  * Bump includes fixes that allow pylint to run in python 3.8
  * Python 3.4 is EOL and not supported by newer versions of pylint, thus has been dropped from travis tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ python:
   - "3.4"
   - "3.5"
   - "3.6"
+  - "3.7"
```
Comment (liam-middlebrook) [inline .travis.yml:5]: Please move into separate commit.

File: requirements-test.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 pycodestyle==2.4.0
-pylint==1.8.4
+pylint==2.5.2
```
Comment (liam-middlebrook) [inline requirements-test.txt:2]: Please move into separate commit.

File: csh_ldap/member.py [hunk 3]
```diff
@@ -90,13 +105,12 @@ def __getattr__(self, key, as_list=False):
...
 
             return ret
-        else:
-            try:
-                return res[0][1][key][0].decode('utf-8')
-            except UnicodeDecodeError:
-                return res[0][1][key][0]
-            except KeyError:
-                return None
+        try:
```
Comment (liam-middlebrook) [inline csh_ldap/member.py:108]: Did pylint complain about this, why was it needed? (Not against it, but it's unrelated to the __repr__ additions and all so I think it would be best in a separate commit.

**Comment by liam-middlebrook** [comment]
Sorry for the slow review. The changes adding new functions here LGTM, but please take a look at my other comments about splitting up commits into granular related pieces.

File: csh_ldap/member.py (hunk 3 above)
Comment (mxmeinhold) [reply csh_ldap/member.py:108]: Yeah pylint complained for else after return.
